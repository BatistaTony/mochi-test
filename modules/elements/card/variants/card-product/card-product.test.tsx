import { render, screen, userEvent } from '@test-utils';

import ProductCard from './index';

jest.mock('uuid', () => {
  let counter = 0;
  return {
    v4: jest.fn().mockImplementation(() => {
      counter += 1;
      return counter;
    }),
  };
});

describe('Product card element testing', () => {
  it('properly renders the mini product card', () => {
    const btnText = 'add to cart';
    const { container } = render(
      <ProductCard
        compressed
        description="tomates 20g, cebola 15g, alface 5g, 1 fatia de pão e molho janonês."
        image={{ small: 'img-300w.png', medium: 'img-600w.png' }}
        productName="nocal"
        price={1000}
        onClick={jest.fn()}
        prepareTime="20min"
        btnText={btnText}
        key="test"
      />,
      { newTheme: true }
    );
    expect(screen.getByText('nocal')).toBeInTheDocument();
    expect(screen.getByAltText('nocal')).toBeInTheDocument();
    expect(screen.getByText('Included:')).toBeInTheDocument();
    expect(screen.queryByText('Time to prepare:')).not.toBeInTheDocument();
    expect(screen.queryByText('Price per unit:')).not.toBeInTheDocument();
    expect(screen.getByRole('button', { name: btnText })).toBeInTheDocument();
    expect(screen.queryByRole('button', { name: /add item/i })).not.toBeInTheDocument();
    expect(screen.queryByRole('button', { name: /remove item/i })).not.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it('properly renders the product card', () => {
    const btnText = 'add to cart';
    const { container } = render(
      <ProductCard
        description="tomates 20g, cebola 15g, alface 5g, 1 fatia de pão e molho janonês."
        image={{ small: 'img-300w.png', medium: 'img-600w.png' }}
        productName="nocal"
        price={1000}
        onClick={jest.fn()}
        prepareTime="20min"
        btnText={btnText}
        key="test"
      />,
      { newTheme: true }
    );
    expect(screen.getByText('nocal')).toBeInTheDocument();
    expect(screen.getByAltText('nocal')).toBeInTheDocument();
    expect(screen.getByText('Included:')).toBeInTheDocument();
    expect(screen.getByText('Time to prepare:')).toBeInTheDocument();
    expect(screen.getByText('Price per unit:')).toBeInTheDocument();
    expect(screen.getByRole('button', { name: btnText })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /add item/i })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /remove item/i })).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it('allows add, remove and add items to cart', () => {
    const onClick = jest.fn();
    const btnText = 'add to cart';
    render(
      <ProductCard
        description="tomates 20g, cebola 15g, alface 5g, 1 fatia de pão e molho janonês."
        image={{ small: 'img-300w.png', medium: 'img-600w.png' }}
        productName="nocal"
        price={1000}
        onClick={onClick}
        prepareTime="20min"
        btnText={btnText}
        key="test"
      />,
      { newTheme: true }
    );
    const addToCartButton = screen.getByRole('button', { name: btnText });
    const incrementButton = screen.getByRole('button', { name: /add item/i });
    const decrementButton = screen.getByRole('button', { name: /remove item/i });

    userEvent.click(addToCartButton);
    expect(onClick).toHaveBeenCalledTimes(1);
    expect(onClick).toHaveBeenCalledWith(1);

    userEvent.click(incrementButton);
    userEvent.click(incrementButton);
    userEvent.click(incrementButton);
    userEvent.click(decrementButton);

    userEvent.click(addToCartButton);
    expect(onClick).toHaveBeenCalledTimes(2);
    expect(onClick).toHaveBeenCalledWith(3);
  });
});
