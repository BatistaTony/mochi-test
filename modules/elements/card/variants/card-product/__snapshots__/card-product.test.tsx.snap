// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product card element testing properly renders the mini product card 1`] = `
.c10 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding: calc(1rem + 0.125rem) 5rem;
  border-radius: 0.937rem;
  line-height: 150%;
  -webkit-letter-spacing: calc(2.1rem * 0.1);
  -moz-letter-spacing: calc(2.1rem * 0.1);
  -ms-letter-spacing: calc(2.1rem * 0.1);
  letter-spacing: calc(2.1rem * 0.1);
  font-weight: bold;
  text-align: center;
  font-size: Medium;
  color: #FFFFFF;
  background-color: #3A0896;
  padding-left: 1rem;
  padding-right: 1rem;
  width: 100%;
}

.c10:hover {
  background-color: #6E43BF;
}

.c10:focus {
  background-color: #3A0896;
}

.c10:active {
  background-color: #745CA0;
}

.c10:disabled {
  background-color: #889BA2;
  cursor: default;
}

.c2 {
  width: 100%;
  height: 100%;
  position: relative;
}

.c2 .lazyload,
.c2 .lazyloading {
  opacity: 0;
}

.c2 .lazyload,
.c2 .lazyloaded,
.c2 {
  opacity: 1;
  -webkit-transition: 2s cubic-bezier(0.215,0.61,0.355,1);
  transition: 2s cubic-bezier(0.215,0.61,0.355,1);
}

.c2:before {
  content: '';
  position: absolute;
  z-index: -1;
  top: 50%;
  left: 50%;
  margin-top: -16px;
  margin-left: -16px;
  width: 32px;
  height: 32px;
  border-radius: 50%;
  -webkit-animation: fTPTbj 1.2s infinite ease-in-out;
  animation: fTPTbj 1.2s infinite ease-in-out;
  background-color: #745CA0;
}

.c2 img {
  position: relative;
  width: 100%;
  max-height: 100%;
  z-index: 0;
  object-fit: contain;
}

.c0 {
  width: 35rem;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  height: 12rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 2rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c6 {
  margin: 2rem 1rem 4rem;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  line-height: 150%;
  font-size: 2.1rem;
  font-weight: 500;
}

.c7 {
  line-height: 150%;
  font-size: 1.6rem;
}

.c8 {
  line-height: 150%;
  font-size: 1.6rem;
  font-weight: bold;
  cursor: pointer;
  color: #273C75;
}

.c8:hover {
  color: #40739e;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c11 {
  line-height: 150%;
  -webkit-letter-spacing: calc(2.1rem * 0.1);
  -moz-letter-spacing: calc(2.1rem * 0.1);
  -ms-letter-spacing: calc(2.1rem * 0.1);
  letter-spacing: calc(2.1rem * 0.1);
  font-weight: bold;
  font-size: 1.2rem;
  padding: 0 0.5rem;
}

.c12 {
  line-height: 150%;
  -webkit-letter-spacing: calc(2.1rem * 0.1);
  -moz-letter-spacing: calc(2.1rem * 0.1);
  -ms-letter-spacing: calc(2.1rem * 0.1);
  letter-spacing: calc(2.1rem * 0.1);
  font-weight: bold;
  font-size: 1.2rem;
  padding: 0 0.5rem;
  text-transform: uppercase;
}

<div>
  <div
    id="__next"
  >
    <div
      aria-label="product"
      class="c0"
      display="flex"
      role="article"
      width="35rem"
    >
      <div
        aria-label="nocal"
        class="c1"
        display="flex"
        height="12rem"
        role="figure"
      >
        <div
          class="c2"
        >
          <img
            alt="nocal"
            class="lazyload"
            data-sizes="auto"
            data-srcset="S3_BUCKET/img-300w.png 300w, S3_BUCKET/img-600w.png 600w"
            src="S3_BUCKET/img-600w.png"
            srcset="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
          />
        </div>
      </div>
      <div
        class="c3"
        display="flex"
      >
        <div
          class="c4"
          display="flex"
        >
          <p
            class="c5"
          >
            nocal
          </p>
          <div
            class="c6"
          >
            <p
              class="c7"
            >
              <span
                class="c7"
              >
                Included: 
              </span>
              tomates 20g, cebola 15g, alface 5g, 1 fatia de pão e molho janonês.
              ...
              <span
                class="c8"
              >
                View more
              </span>
            </p>
          </div>
        </div>
        <div
          class="c9"
          display="flex"
          width="100%"
        >
          <button
            aria-labelledby="6"
            class="c10"
            type="button"
            width="100%"
          >
            <p
              aria-label="price"
              class="c11"
            >
              1000
            </p>
            <p
              class="c12"
              id="6"
            >
              add to cart
            </p>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Product card element testing properly renders the product card 1`] = `
.c9 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding: calc(1rem + 0.125rem) 5rem;
  border-radius: 0.937rem;
  line-height: 150%;
  -webkit-letter-spacing: calc(2.1rem * 0.1);
  -moz-letter-spacing: calc(2.1rem * 0.1);
  -ms-letter-spacing: calc(2.1rem * 0.1);
  letter-spacing: calc(2.1rem * 0.1);
  font-weight: bold;
  text-align: center;
  font-size: Medium;
  color: #FFFFFF;
  background-color: #3A0896;
  padding-left: 1rem;
  padding-right: 1rem;
  width: 100%;
}

.c9:hover {
  background-color: #6E43BF;
}

.c9:focus {
  background-color: #3A0896;
}

.c9:active {
  background-color: #745CA0;
}

.c9:disabled {
  background-color: #889BA2;
  cursor: default;
}

.c13 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding: calc(1rem + 0.125rem) 5rem;
  border-radius: 0.937rem;
  line-height: 150%;
  -webkit-letter-spacing: calc(2.1rem * 0.1);
  -moz-letter-spacing: calc(2.1rem * 0.1);
  -ms-letter-spacing: calc(2.1rem * 0.1);
  letter-spacing: calc(2.1rem * 0.1);
  font-weight: bold;
  text-align: center;
  font-size: Medium;
  padding: 0;
  font-weight: normal;
  color: #FFFFFF;
  background-color: #3A0896;
  width: calc(1rem * 3);
  height: calc(1rem * 3);
  border-radius: 50%;
  margin: 0 0.25rem;
}

.c13:hover {
  background-color: #6E43BF;
}

.c13:focus {
  background-color: #3A0896;
}

.c13:active {
  background-color: #745CA0;
}

.c13:disabled {
  background-color: #889BA2;
  cursor: default;
}

.c2 {
  width: 100%;
  height: 100%;
  position: relative;
}

.c2 .lazyload,
.c2 .lazyloading {
  opacity: 0;
}

.c2 .lazyload,
.c2 .lazyloaded,
.c2 {
  opacity: 1;
  -webkit-transition: 2s cubic-bezier(0.215,0.61,0.355,1);
  transition: 2s cubic-bezier(0.215,0.61,0.355,1);
}

.c2:before {
  content: '';
  position: absolute;
  z-index: -1;
  top: 50%;
  left: 50%;
  margin-top: -16px;
  margin-left: -16px;
  width: 32px;
  height: 32px;
  border-radius: 50%;
  -webkit-animation: fTPTbj 1.2s infinite ease-in-out;
  animation: fTPTbj 1.2s infinite ease-in-out;
  background-color: #745CA0;
}

.c2 img {
  position: relative;
  width: 100%;
  max-height: 100%;
  z-index: 0;
  object-fit: contain;
}

.c0 {
  width: 45rem;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  height: 12rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 2rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c6 {
  margin: 2rem 1rem 4rem;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c12 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c5 {
  line-height: 150%;
  font-size: 2.1rem;
  font-weight: 500;
}

.c7 {
  line-height: 150%;
  font-size: 1.6rem;
}

.c10 {
  line-height: 150%;
  -webkit-letter-spacing: calc(2.1rem * 0.1);
  -moz-letter-spacing: calc(2.1rem * 0.1);
  -ms-letter-spacing: calc(2.1rem * 0.1);
  letter-spacing: calc(2.1rem * 0.1);
  font-weight: bold;
  font-size: 1.2rem;
  padding: 0 0.5rem;
}

.c11 {
  line-height: 150%;
  -webkit-letter-spacing: calc(2.1rem * 0.1);
  -moz-letter-spacing: calc(2.1rem * 0.1);
  -ms-letter-spacing: calc(2.1rem * 0.1);
  letter-spacing: calc(2.1rem * 0.1);
  font-weight: bold;
  font-size: 1.2rem;
  padding: 0 0.5rem;
  text-transform: uppercase;
}

<div>
  <div
    id="__next"
  >
    <div
      aria-label="product"
      class="c0"
      display="flex"
      role="article"
      width="45rem"
    >
      <div
        aria-label="nocal"
        class="c1"
        display="flex"
        height="12rem"
        role="figure"
      >
        <div
          class="c2"
        >
          <img
            alt="nocal"
            class="lazyload"
            data-sizes="auto"
            data-srcset="S3_BUCKET/img-300w.png 300w, S3_BUCKET/img-600w.png 600w"
            src="S3_BUCKET/img-600w.png"
            srcset="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
          />
        </div>
      </div>
      <div
        class="c3"
        display="flex"
      >
        <div
          class="c4"
          display="flex"
        >
          <p
            class="c5"
          >
            nocal
          </p>
          <div
            class="c6"
          >
            <p
              class="c7"
            >
              <span
                class="c7"
              >
                <b>
                  Included: 
                </b>
              </span>
              tomates 20g, cebola 15g, alface 5g, 1 fatia de pão e molho janonês.
               
            </p>
            <p
              class="c7"
            >
              <span
                class="c7"
              >
                <b>
                  Time to prepare: 
                </b>
              </span>
              20min
            </p>
            <p
              class="c7"
            >
              <span
                class="c7"
              >
                <b>
                  Price per unit: 
                </b>
              </span>
              1000
            </p>
          </div>
        </div>
        <div
          class="c8"
          display="flex"
          width="100%"
        >
          <button
            aria-labelledby="12"
            class="c9"
            type="button"
            width="100%"
          >
            <p
              aria-label="price"
              class="c10"
            >
              1000
            </p>
            <p
              class="c11"
              id="12"
            >
              add to cart
            </p>
          </button>
          <div
            class="c12"
            display="inline-flex"
          >
            <button
              aria-disabled="true"
              aria-labelledby="10"
              class="c13"
              disabled=""
              type="button"
            >
              <svg
                aria-label="remove item"
                fill="currentColor"
                height="1em"
                id="10"
                role="img"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                />
              </svg>
            </button>
            <button
              aria-disabled="true"
              aria-label="quantity"
              class="c13"
              color="text"
              disabled=""
              type="button"
            >
              1
            </button>
            <button
              aria-labelledby="11"
              class="c13"
              type="button"
            >
              <svg
                aria-label="add item"
                fill="currentColor"
                height="1em"
                id="11"
                role="img"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
